numpy/random/mtrand/mtrand.pyx

```diff
@@ -4901,10 +4901,13 @@ cdef class RandomState:
         """
         if isinstance(x, (int, long, np.integer)):
             arr = np.arange(x)
-        else:
-            arr = np.array(x)
-        self.shuffle(arr)
-        return arr
+            self.shuffle(arr)
+            return arr
+
+        arr = np.array(x)
```

Comment:
This should be `asarray`, since you're not modifying it in this case.

Reply:
I added this but then needed to check when using the fast path